### PR TITLE
perl Test::* modules not part of Perl's extensions

### PR DESCRIPTION
**Pull request by xaver42**
Maybe these could even go into p/Perl/Perl-5.16.3-goolf-1.4.10.eb ?

### CODE REVIEW

**Comment by xaver42** [comment]
@boegel: is there anything I can do to have this included in the next release?

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/t/Test-Warnings/Test-Warnings-0.020-goolf-1.4.10-Perl-5.16.3.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+source_urls = ['http://www.cpan.org/modules/by-module/Test']
+sources = [SOURCE_TAR_GZ]
+
+perl = 'Perl'
+perlver = '5.16.3'
+perlverextra = ''
+versionsuffix = '-%s-%s%s' % (perl, perlver, perlverextra)
+
+dependencies = [
+    (perl,        perlver, perlverextra),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Test-Warnings/Test-Warnings-0.020-goolf-1.4.10-Perl-5.16.3.eb]: please fix the spacing here:

```python
    (perl, perlver, perlverextra),
```

File: easybuild/easyconfigs/t/Test-Number-Delta/Test-Number-Delta-1.06-goolf-1.4.10-Perl-5.16.3.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = ['http://www.cpan.org/modules/by-module/Test']
+sources = [SOURCE_TAR_GZ]
+
+perl = 'Perl'
+perlver = '5.16.3'
+perlverextra = ''
+versionsuffix = '-%s-%s%s' % (perl, perlver, perlverextra)
+
+dependencies = [
+    (perl,        perlver, perlverextra),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Test-Number-Delta/Test-Number-Delta-1.06-goolf-1.4.10-Perl-5.16.3.eb]: fix spacing please

File: easybuild/easyconfigs/t/Test-Cmd/Test-Cmd-1.08-goolf-1.4.10-Perl-5.16.3.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+source_urls = ['http://www.cpan.org/modules/by-module/Test']
+sources = [SOURCE_TAR_GZ]
+
+perl = 'Perl'
+perlver = '5.16.3'
+perlverextra = ''
+versionsuffix = '-%s-%s%s' % (perl, perlver, perlverextra)
+
+dependencies = [
+    (perl,        perlver, perlverextra),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Test-Cmd/Test-Cmd-1.08-goolf-1.4.10-Perl-5.16.3.eb]: fix spacing please

**Comment by boegel** [comment]
@xaver42: thanks for bumping this back into view, it went under my radar

Other than the minor style remarks, it looks good. I'll look into submitting a test report for it.

Just a suggestion (up to you what you do with it): it seems like these three are closely related, so one other/better option may be to collapse them into a single 'bundle' easyconfig. See for example https://github.com/hpcugent/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/r/R-bundle-Bioconductor.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 3 (3 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/cf56a884e8f064bececd for a full test report.

**Comment by boegel** [comment]
@xaver42: since EasyBuild v2.3.0, the fallback sanity check paths (`bin` + `lib` or `lib64`) are also checked for extensions-type easyconfigs (like Perl modules) if no custom `sanity_check_paths` are provided

so, you'll need to include a custom definitions for `sanity_check_paths` in each of these to make it pass the sanity check

see also https://lists.ugent.be/wws/arc/easybuild/2015-09/msg00024.html

**Comment by xaver42** [comment]
@boegel ok, it makes sense to have `sanity_check_paths` set. Regarding the bundle: Would it make sense to include these tests in `p/Perl/Perl-5.16.3-goolf-1.4.10.eb`? It already contains some of the Test packages.

**Comment by boegel** [comment]
@xaver42: sure, but then they should, ideally, be added in all other (non-bare) Perl easyconfigs too (which is only a bit of a burden for testing this PR, but that's my problem)

adding additional extensions to `exts_list` is fine

**Comment by xaver42** [comment]
@boegel this apparently hasn't been done in the past - `Test::Warnings` version `0.014` is part of the non-bare versions of `Perl-5.20.0*`. I guess upgrading to version `0.020` (or even version `0.021` which came out after my PR) is not an option for these? Btw, I just noted that the `source_urls` with `http://www.cpan.org/modules/by-module` seem to contain only the last few versions of a module (as opposed to `http://cpan.metacpan.org/authors/*`), so this may become a problem in the future (currently used by `DB_File`, `Libwww-Perl` and `YAML-Syck`, the latter a PR from myself some time ago).

**Comment by boegel** [comment]
@xaver42: we stay away from updating extension versions, yes, in that case you should go with a separate (bundle) module

there's little we can do about sources that disappear upstream, other than archiving things ourselves somewhere; there have been discussions on that, but it's tricky business w.r.t. licensing etc.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
node2098.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/702d25a2c74093e18482 for a full test report.

**Comment by migueldiascosta** [comment]
@xaver42, this PR is being closed for the following reason(s): no activity for > 6 months.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**